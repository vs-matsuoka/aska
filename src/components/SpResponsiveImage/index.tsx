import Image, { ImageProps } from 'next/image';

function SpResponsiveImage(
  props: Omit<ImageProps, 'fill' | 'width' | 'height'> & {
    width: number;
    height: number;
  }
) {
  const { width, height, src, alt, className, ...imageProps } = props;
  const standard = 750;

  if (width === standard) {
    return <Image src={src} alt={alt} width={width} height={height} {...imageProps} />;
  }

  return (
    <div
      className={className}
      style={{
        width: `${(width / standard) * 100}vw`,
        height: `${(height / standard) * 100}vw`
      }}
    >
      <Image src={src} alt={alt} className="max-w-none" fill {...imageProps} />
    </div>
  );
}

export default SpResponsiveImage;
