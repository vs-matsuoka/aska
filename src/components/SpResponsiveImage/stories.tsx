import { Story, Meta } from '@storybook/react';
import SpResponsiveImage from '.';

export default {
  title: 'SpResponsiveImage',
  component: SpResponsiveImage
} as Meta;

export const Default: Story = () => <SpResponsiveImage src="/Entry/21_Entry_text_02_shadow.png" width={480} height={105} alt="a" />;
