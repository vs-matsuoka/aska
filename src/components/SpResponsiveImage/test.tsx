import { render, screen } from '@testing-library/react';

import SpResponsiveImage from '.';

describe('<SpResponsiveImage />', () => {
  it('should render the heading', () => {
    const { container } = render(<SpResponsiveImage src="/Entry/21_Entry_text_02_shadow.png" width={480} height={105} alt="a" />);

    expect(container.querySelector('div')?.style.width).toBe(`${(480 / 750) * 100}vw`);
    expect(container.querySelector('div')?.style.height).toBe(`${(105 / 750) * 100}vw`);

    expect(container.firstChild).toMatchSnapshot();
  });
});
