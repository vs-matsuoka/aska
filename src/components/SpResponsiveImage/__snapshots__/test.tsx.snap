// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SpResponsiveImage /> should render the heading 1`] = `
<div
  style="width: 64vw; height: 14.000000000000002vw;"
>
  <img
    alt="a"
    class="max-w-none"
    data-nimg="future-fill"
    decoding="async"
    loading="lazy"
    sizes="100vw"
    src="/_next/image?url=%2FEntry%2F21_Entry_text_02_shadow.png&w=3840&q=75"
    srcset="/_next/image?url=%2FEntry%2F21_Entry_text_02_shadow.png&w=640&q=75 640w, /_next/image?url=%2FEntry%2F21_Entry_text_02_shadow.png&w=750&q=75 750w, /_next/image?url=%2FEntry%2F21_Entry_text_02_shadow.png&w=828&q=75 828w, /_next/image?url=%2FEntry%2F21_Entry_text_02_shadow.png&w=1080&q=75 1080w, /_next/image?url=%2FEntry%2F21_Entry_text_02_shadow.png&w=1200&q=75 1200w, /_next/image?url=%2FEntry%2F21_Entry_text_02_shadow.png&w=1920&q=75 1920w, /_next/image?url=%2FEntry%2F21_Entry_text_02_shadow.png&w=2048&q=75 2048w, /_next/image?url=%2FEntry%2F21_Entry_text_02_shadow.png&w=3840&q=75 3840w"
    style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
  />
</div>
`;
